lib/dataTableHelpers.tsx

```diff
@@ -23,3 +23,74 @@ export function truncatedTableCell(file: Entity) {
     const value = this.selector(file);
     return <ExpandableText fullText={value} truncateProps={{ lines: 4 }} />;
 }
+
+interface ISimpleColumnVisibilityDef {
+    name: string;
+    id?: string;
+    visible?: boolean;
+}
+
+export function resolveColumnVisibilityByColumnDefinition(
+    columns: ISimpleColumnVisibilityDef[] = []
+): { [columnId: string]: boolean } {
+    const colVis: { [columnId: string]: boolean } = {};
+
+    columns.forEach((column: ISimpleColumnVisibilityDef) => {
+        // every column is visible by default unless it is flagged otherwise
+        let visible = true;
+
+        if (column.visible !== undefined) {
+            visible = column.visible;
+        }
+
+        // if no id exists, just use name for key
+        const key = column.id || column.name;
+
+        colVis[key] = visible;
+    });
+
+    return colVis;
+}
+
+export function resolveColumnVisibility(
+    columnVisibilityByColumnDefinition: { [columnId: string]: boolean },
+    columnVisibility?: { [columnId: string]: boolean },
+    columnVisibilityOverride?: { [columnId: string]: boolean }
```

Comment:
`columnVisibilityUnsubmitted`?

Reply:
not sure if i understand either.   if it's user selection then it probably should mention that in the name.  like userSelectedColumnVisibility?